maven-central-publish/src/main/java/tech/yanand/gradle/PublishToCentralPortalTask.java

```diff
@@ -87,11 +92,19 @@ void executeTask() throws InterruptedException, IOException {
                 throw apiNotReturnDeploymentStateField();
             } else if (FAILED.equals(deploymentStatus)) {
                 throw deploymentStatusIsField(deploymentStatus);
-            } else if (PUBLISHING.equals(deploymentStatus) || PUBLISHED.equals(deploymentStatus)) {
-                getLogger().lifecycle("Upload file success! current status: {}.", deploymentStatus);
-                return;
             } else {
-                ++count;
+                boolean isSuccessful = PUBLISHING.equals(deploymentStatus) || PUBLISHED.equals(deploymentStatus);
+
+                if (currentPublishingType == PublishingType.USER_MANAGED) {
+                    isSuccessful = isSuccessful || VALIDATED.equals(deploymentStatus);
+                }
+
+                if (isSuccessful) {
+                    getLogger().lifecycle("Upload file success! current status: {}.", deploymentStatus);
+                    return;
+                } else {
+                    ++count;
+                }
```

Comment:
Can we extract a method `isSuccessful()` here to make it look clearer?

```java
else if (FAILED.equals(deploymentStatus)) {
    throw deploymentStatusIsField(deploymentStatus);
} else if (isSuccessful(deploymentStatus)) {
    getLogger().lifecycle("Upload file success! current status: {}.", deploymentStatus);
    return;
} else {
    ++count;
}


private boolean isSuccessful(String deploymentStatus) {
	return (publishingType.get() == PublishingType.USER_MANAGED && VALIDATED.equals(deploymentStatus))
		|| PUBLISHING.equals(deploymentStatus)
		|| PUBLISHED.equals(deploymentStatus)
}

```

Reply:
Done, thank you!